harness/main.go

```diff
@@ -39,5 +40,5 @@ func MakeApp(opts *ExporterOpts) *cli.App {
 }
 
 func Main(opts *ExporterOpts) {
-	MakeApp(opts).RunAndExitOnError()
+	MakeApp(opts).Run(os.Args)
```

Comment:
So the reason I used `RunAndExitOnError` here is to let it exit with error when a user specified invalid command line argument such as below:

```
~/prometheus-exporter-harness
$ git co master                        
Already on 'master'
Your branch is up-to-date with 'origin/master'.

~/prometheus-exporter-harness
$ ./example_exporter --flagfoo=hogheoge >/dev/null 2>&1 # Wrong usage as the flagfoo is a boolean flag

~/prometheus-exporter-harness
$ echo $?
1
```

In order to keep this behavior, I think you have to handle the return value from `Run()`; an error returned from cli package if any.